cardano-testnet/src/Testnet/Components/Query.hs

```diff
@@ -77,26 +75,27 @@ waitUntilEpoch nodeConfigFile socketPath desiredEpoch = withFrozenCallStack $ do
         <> "- invalid foldEpochState behaviour, result: " <> show res
       H.failure
 
+-- | Wait for the number of epochs
+waitForEpochs
+  :: MonadTest m
+  => MonadCatch m
+  => MonadIO m
+  => ExecConfig
+  -> NodeConfigFile In
+  -> SocketPath
+  -> EpochInterval  -- ^ Number of epochs to wait
+  -> m EpochNo -- ^ The epoch number reached
+waitForEpochs execConfig nodeConfigFile socketPath interval = withFrozenCallStack $ do
+  currentEpoch <- H.nothingFailM $ mEpoch <$> queryTip execConfig
```

Comment:
So I don't think `NewEpochState` has the current `EpochNo` but it may be worth asking Alexy.

Reply:
Is this comment still relevant?

Reply:
Will do in a follow-up PR.